src/HH4b/postprocessing/postprocessing.py

```diff
@@ -518,13 +542,39 @@ def get_templates(
                 # add weight variations
                 for wshift, wsyst in weight_shifts.items():
                     if sample in wsyst.samples and year in wsyst.years:
-                        for skey, shift in [("Down", "down"), ("Up", "up")]:
-                            # reweight based on diff between up/down and nominal weights
-                            h.fill(
-                                Sample=f"{sample}_{wshift}_{shift}",
-                                **fill_data,
-                                weight=events[f"weight_{wshift}{skey}"].to_numpy().squeeze(),
-                            )
+                        if wshift not in ["scale", "pdf"]:
+                            # fill histogram with weight variations
+                            for skey, shift in [("Down", "down"), ("Up", "up")]:
+                                h.fill(
+                                    Sample=f"{sample}_{wshift}_{shift}",
+                                    **fill_data,
+                                    weight=events[f"weight_{wshift}{skey}"].to_numpy().squeeze(),
+                                )
+                        else:
+                            # get histograms for all QCD scale and PDF variations
+                            whists = _get_qcdvar_hists(events, shape_vars, fill_data, wshift)
+
+                            if wshift == "scale":
+                                # renormalization / factorization scale uncertainty is the max/min envelope of the variations
+                                shape_up = np.max(whists.values(), axis=0)
+                                shape_down = np.min(whists.values(), axis=0)
+                            else:
+                                # pdf uncertainty is the norm of each variation (corresponding to 103 eigenvectors) - nominal
+                                nom_vals = h[sample, :].values()
+                                abs_unc = np.linalg.norm(
+                                    (whists.values() - nom_vals), axis=0
+                                )  # / np.sqrt(103)
```

Comment:
why is `# / np.sqrt(103)` commented? maybe get rid of it?

Reply:
Yes it is not necessary (wasn't sure before).